### ISSUE TITLE
Different result of cross entropy loss between result from model as function and using predict function

### ISSUE THREAD

**Issue by Xinshuai-Lyu** [tensorflow/tensorflow#56555]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Bug

### Source

source

### Tensorflow Version

2.8.2

### Custom Code

Yes

### Current Behaviour?

```shell
Firstly, you get a prediction using model as a function, call this predition A; and then you use predict function of the model and get the same seemingly same prediction, call it B.

If you use CategoricalCrossentropy to calculate loss with same label, the loss are different between them.
```

### Standalone code to reproduce the issue

```shell
class NLPModel(keras.Model):
  def __init__(self, **kwargs):
    super().__init__(**kwargs)
    self.hidden_layer1 = tf.keras.layers.GRU(400)
    self.outputs = tf.keras.layers.Dense(3, activation="sigmoid")
  def call(self, inputs):
    outputs = self.hidden_layer1(inputs)
    outputs = self.outputs(outputs)
    return outputs
model = NLPModel()
label = tf.constant([[1.,0.,0.]])
prediction = model(tf.constant([[[0.4, 0.5, 0.54,0.4, 0.5, 0.54,0.4, 0.5, 0.54,0.9],
                                  [0.4, 0.5, 0.54,0.4, 0.5, 0.54,0.4, 0.5, 0.54,0.9]]]))
cce = tf.keras.losses.CategoricalCrossentropy()
loss = cce(label, prediction)
def custom_cce(label, prediction):
  return tf.reduce_mean(tf.math.log(tf.matmul(label, tf.transpose(prediction)) / tf.reduce_sum(prediction)))
print("before numpy")
print(prediction)
print(loss)
print("before numpy")
print("after custom_cce")
print(custom_cce(label, prediction))
print("after custom_cce")
prediction = prediction.numpy()
loss = cce(label, prediction)
print("after numpy")
print(prediction)
print(loss)
print("after numpy")
```

### Relevant log output

```shell
before numpy
tf.Tensor([[0.5021686  0.48505145 0.5026401 ]], shape=(1, 3), dtype=float32)
tf.Tensor(1.0769439, shape=(), dtype=float32)
before numpy
after custom_cce
tf.Tensor(-1.0875016, shape=(), dtype=float32)
after custom_cce
after numpy
[[0.5021686  0.48505145 0.5026401 ]]
tf.Tensor(1.0875016, shape=(), dtype=float32)
after numpy
```
</details>

**Comment by sushreebarsa** [comment]
@chunduriv I was able to replicate the issue on colab, please find the gist [here](https://colab.research.google.com/gist/sushreebarsa/089d2bddcaa2c1dd06d8493917583273/56555.ipynb#scrollTo=wy0ahziJZcWC) for reference. Thank you!

**Comment by chunduriv** [comment]
@Xinshuai-Lyu,

Thank you for opening this issue. 

Development of keras moved to separate repository https://github.com/keras-team/keras/issues. 

Please post this issue on `keras-team/keras repo`. For more details please refer [here](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999).

**Comment by Xinshuai-Lyu** [comment]
https://github.com/keras-team/keras/issues/16737
